lib/AppConfig.php

```diff
@@ -89,7 +89,19 @@ public function setUserValue($userId, $key, $value) {
 	 * @return bool
 	 */
 	public function enterpriseFeaturesEnabled() {
-		if (!$this->licenseManager->checkLicenseFor($this->appName)) {
+		$isEnterprise = $this->config->getAppValue($this->appName, 'isEEActive', null);
+		if ($isEnterprise === null) {
+			// app not marked as EE -> return false without doing anything more
+			return false;
+		}
+
+		$isEnterprise = \filter_var($isEnterprise, FILTER_VALIDATE_BOOLEAN);
+		if (!$isEnterprise || !$this->licenseManager->checkLicenseFor($this->appName)) {
+			// if not marked as enterprise, no need to check the license:
+			// explicitly mark the app as non-enterprise and remove the secure view.
+			// if marked as enterprise, we need to verify the license
+			$this->config->setAppValue($this->appName, 'isEEActive', 'no');
+			$this->config->setAppValue($this->appName, 'secure_view_option', 'false');
 			return false;
 		}
 
```

Comment:
what happens in case of true? Wont we need to set app values in this case too ?

Reply:
If there is no "isEEActive" key, the code won't reach this point. The app will be considered as CE.
The admin will need to explicitly set the key via the `config:app:set` command.
If he set the value as "yes", we'll check the license and revert the value if the license is wrong. We'll also revert the "secure_view_option" because the app is considered now as CE due to the wrong license.
If he set the value as "yes" and the license is valid, we don't need to change anything. The admin must have changed the "isEEActive" value manually, so we don't need to change it, and we also don't want to modify the other flags such as "secure_view_option" (the app could be EE but secure_view might not be active)

Reply:
I still am blury. Can you write below flow in points: - enabling richdocuments when having enterprise license, - enabling richdocuments not having the license

Reply:
* No "isEEActive" key => do nothing. Only community features **should** enabled (upgrade is a special case)
* "isEEActive" = "yes":
  * if valid license => enable EE features
  * if invalid license => set "isEEActive" = "no" and disable EE features (secure_view will be false)
* "isEEActive" = "no" => set "isEEActive" = "no" and disable EE features

For the upgrade case, QA should verify that it works as EE if there is a valid license. This scenario can fall into
* "isEEActive" = "yes" + valid license (as instructed, no problems expected)
* no "isEEActive" key (need to ensure nothing wrong happens)

In this last scenario, I've noticed that there is a "license is required" message for the secure_view, but it still shows as active if it was active before, as well as the rest of the related options. In case it wasn't active it's possible that the checkbox remains disabled. This is fixed by setting the "isEEActive" = "yes"

Reply:
It is not fully my question. I wamt to know what *system-admin* has to do given just TAR with everything, starting from scratch.

Reply:
For a fresh installation, the admin won't have to do anything if he just want to use CE features. If he want to use EE features, he'll need to provide a valid license and set the "isEEActive" = "yes".

In case of an upgrade, if he wants CE features, he won't need to do anything. If the current version has EE features enabled, he should run the occ command before the upgrade

Reply:
@pmaier is this acceptable? To me sounds fishy.